migrations/20170103182459_data_categories_remapping.js

```diff
@@ -0,0 +1,144 @@
+    group: 'Data' },
+
+  { name: 'Blog about trial design or results',
+    group: 'Miscellaneous' },
+  { name: 'Journal article critiquing trial design or results',
+    group: 'Miscellaneous',
+    replaces: ['Critical review'] },
+  { name: 'Systematic review including trial',
+    group: 'Miscellaneous' },
+  { name: 'Review article citing trial',
+    group: 'Miscellaneous' },
+  { name: 'News article about trial or results',
+    group: 'Miscellaneous' },
+  { name: 'Press release about trial',
+    group: 'Miscellaneous' },
+  { name: 'Report from sponsor describing trial or results',
+    group: 'Miscellaneous' },
+];
+
+const toInsert = _.map(newCategories, (obj) => _.omit(obj, ['replaces']));
+const toReplace = _.filter(newCategories, 'replaces');
+const toDelete = _.reject(newCategories, (obj) => obj.replaces);
+
+exports.up = (knex) => knex
+  .batchInsert('data_categories', toInsert, 50)
+  .then(() =>
+        _.map(toReplace, (category) =>
+              category.replaces.forEach((oldName) => knex
+                                        .raw(`UPDATE data_contributions
+                                              SET data_category_id = (SELECT id FROM data_categories
+                                                            WHERE name = '${category.name}'
+                                                            AND "group" = '${category.group}')
+                                              WHERE data_category_id = (SELECT id FROM data_categories
+                                                            WHERE name = '${oldName}'
+                                                            AND "group" IS NULL);`)
+                                        .then(() => knex
+                                              .raw(`DELETE FROM data_categories
+                                                    WHERE name = '${oldName}' AND "group" IS NULL;`))
+        )
+      )
+    );
+
+exports.down = (knex) => {
+  const toRestore = [
+    { name: 'Consent forms', group: null },
+    { name: 'Publication (press release, article, blog post, etc.)', group: null },
+    { name: 'Condition', group: null },
+    { name: 'Intervention', group: null },
+  ];
+
+  return knex.batchInsert('data_categories', toRestore, 50)
+    .then(() => {
+      const replaceAction = _.map(toReplace, (category) =>
+                                  knex('data_categories')
+                                  .where({ name: category.name, group: category.group })
+                                  .update({ name: category.replaces[0] })
+                                  .then()
+      );
+
+      const deleteAction = _.map(toDelete, (category) =>
+                                 knex('data_categories')
+                                 .where({ name: category.name, group: category.group })
+                                 .del()
+                                 .then()
+      );
+
+      return replaceAction && deleteAction;
```

Comment:
After rollback there are `data_contributions` that reference `data_categories` that no longer exist. Please make sure you also update foreign keys to `data_categories` in `down` or set them to null for categories that no longer exist.